Makefile.include

```diff
@@ -512,15 +512,23 @@ BUILDDEPS += $(RIOTBUILD_CONFIG_HEADER_C)
 BUILDDEPS += pkg-prepare
 BUILDDEPS += $(APPDEPS)
 
+# Randomize library order to protect against code re-use attacks (link time reordering)
+RIOTLINKSEED ?= -1
+ifeq ($(BUILDOSXNATIVE),1)
+  SHUFLIBS = $(shell find $(BASELIBS) -size +8c | $(RIOTTOOLS)/randsort/randsort.py --seed $(RIOTLINKSEED))
```

Comment:
Can we avoid calling out to python? On Linux, there should be "shuf". OS X doesn't have that (by default). I think something could be built with "jot | ...".

Or we start requiring some homebrew tools on OS X...

Reply:
I was under the impression that python was required anyhow. I would try to avoid using non-POSIX utilities such as shuf. Besides, those utilities may not allow specifying a seed argument and may not use cryptographic secure random sources.

> Or we start requiring some homebrew tools on OS X...

Why prefer a different third-party utility over the randsort script which doesn't have any external dependencies apart from python itself?

Reply:
It might be reqiured, but should be avoided.
python is slow. e.g.:

```
$ time (echo -n  foo bar 1234 | python3 randsort.py)
foo bar 1234
real    0m0.418s
user    0m0.383s
sys     0m0.039s
```

That's on a raspi3. Not the fastest, but that's *almost half a second* for shuffling three values.

Reply:
I can rewrite it in C if performance is an issue.

Reply:
I just thought of this again and the fact that takes half a second for shuffling three values is likely attributable to the initialization of the python interpreter. As such, shuffling more values should not take significantly longer. Besides, half a second is not really that big a deal for a build process in my opinion.

Reply:
How about using `shuf` when available and using `randsort.py` as a fallback?

Reply:
Why? This would just unnecessarily complicate the entire setup. The `randsort.py` script doesn't have any disadvantages unless you are trying to claim that half a second is a noticeable build slowdown.

Reply:
Half a second is not a noticeable slowdown, but [CI](https://ci.riot-os.org/) currently does 58746 builds for each PR.